apps/www/_blog/2023-11-01-supabase-is-now-compatible-with-nextjs-14.mdx

```diff
@@ -0,0 +1,192 @@
+
+Therefore, we need to configure `supabase-js` to use cookies instead of `localStorage` when running on the server. But this code is a little verbose to ask people to duplicate across every app they build with Supabase:
+
+```tsx
+const supabase = createClient(supabaseUrl, supabaseAnonKey, {
+  auth: {
+    flowType: 'pkce',
+    autoRefreshToken: false,
+    detectSessionInUrl: false,
+    persistSession: true,
+    storage: {
+      getItem: async (key: string) => {
+        cookieStore.get(key)
+      },
+      setItem: async (key: string, value: string) => {
+        cookieStore.set(key, value)
+      },
+      removeItem: async (key: string) => {
+        cookieStore.remove(key)
+      },
+    },
+  },
+})
+```
+
+That takes care of the server-side pieces of Next.js, but since we recommend securing your apps with [Row Level Security (RLS) policies](https://www.youtube.com/watch?v=Ow_Uzedfohk), you can safely access your user's session client-side too. Therefore, we need to tell the browser how access that cookie too:
+
+```tsx
+const supabase = createClient(supabaseUrl, supabaseAnonKey, {
+  auth: {
+    flowType: 'pkce',
+    autoRefreshToken: true,
+    detectSessionInUrl: true,
+    persistSession: true,
+    storage: {
+      getItem: async (key: string) => {
+        return parse(document.cookie[key])
+      },
+      setItem: async (key: string, value: string) => {
+        document.cookie = serialize(key, value)
+      },
+    },
+    removeItem: async (key: string) => {
+      document.cookie = serialize(key, '', {
+        maxAge: 0,
+      })
+    },
+  },
+})
+```
+
+That is a lot of very confusing code! So we decided to [create a package called `@supabase/ssr`](https://supabase.com/docs/guides/auth/server-side/overview) that does all of this for you. Then we took it one step further and created a [Next.js and Supabase Starter Template](https://supabase.com/docs/guides/getting-started/quickstarts/nextjs), so you can just focus on building your awesome app! 🚀
+
+> Check out my [Next.js Conf talk](https://www.youtube.com/watch?t=3880&v=FdiX5rHS_0Y) to see this starter template in action!
+
+## How do I get started?
+
+One command:
+
+```bash
+npx create-next-app@latest -e with-supabase
+```
+
+The landing page will guide you through the process of creating a Supabase project and configuring your environment variables.
+
+Build ~~in a weekend~~ on a Friday night! Scale to millions!
+
+## Meme zone
+
+No Supabase blog is complete without a meme signing off. This one is of the least controversial command coming out of Next.js Conf:
+
+<img
+  className="dark:hidden !m-0"
+  alt="Tweet from Terry Sutton running the `with-supabase` command - light mode version"
```

Comment:
No link!?   :^)

Reply:
I can only link the dark-mode version. It fails the build with an error on light mode....

...not sure why!